packages/core/src/actions/execute.ts

```diff
@@ -1137,6 +1137,62 @@ export const sortSigners = (arr: Array<TreeSigner>): void => {
   })
 }
 
+/**
+ * This is a utility function that is helpful for recovering from a deployment that has timed out. It fetches
+ * all the transaction receipts related to a deployment using the expected events and returns them. If this
+ * function is used when some of the transaction receipts are already tracked, they can be passed in and this
+ * function will return an array of all of the unique transactions.
+ *
+ * This function returns standard ethers.TransactionReceipt objects instead of our SphinxTransactionReceipt
+ * type because that is more flexible and generally useful.
+ */
+export const fetchExecutionTransactionReceipts = async (
+  receipts: ethers.TransactionReceipt[],
+  moduleAddress: string,
+  merkleRoot: string,
+  provider: SphinxJsonRpcProvider | HardhatEthersProvider
+) => {
+  const module = new ethers.Contract(moduleAddress, SphinxModuleABI, provider)
+
+  const SphinxMerkleRootApprovedFilter =
+    module.filters.SphinxMerkleRootApproved(merkleRoot)
+  const SphinxMerkleRootCanceledFilter =
+    module.filters.SphinxMerkleRootCanceled(merkleRoot)
+  const SphinxMerkleRootFailedFilter =
+    module.filters.SphinxMerkleRootFailed(merkleRoot)
+  const SphinxActionSucceededFilter =
+    module.filters.SphinxActionSucceeded(merkleRoot)
+  const SphinxActionFailedFilter = module.filters.SphinxActionFailed(merkleRoot)
+
+  const filters = [
+    SphinxMerkleRootApprovedFilter,
+    SphinxMerkleRootCanceledFilter,
+    SphinxMerkleRootFailedFilter,
+    SphinxActionSucceededFilter,
+    SphinxActionFailedFilter,
+  ]
+
+  const txHashes = receipts.map((r) => r.hash)
+
+  for (const filter of filters) {
+    const latestBlock = await provider.getBlockNumber()
```

Comment:
Seems like we should put `await provider.getBlockNumber` outside of the for-loop so that we only call it once instead of 5 times